src/bonuses.cpp

```diff
@@ -0,0 +1,275 @@
+#include "bonuses.h"
+#include "damage.h"
+#include "json.h"
+#include "character.h"
+#include "debug.h"
+#include <map>
+#include <string>
+#include <utility>
+#include <sstream>
+#include <vector>
+
+#define dbg(x) DebugLog((DebugLevel)(x),D_MAIN) << __FILE__ << ":" << __LINE__ << ": "
+
+// Tokenize
```

Comment:
what? why? The thing has already been run through a json tokenizer, why are you adding another one?
You can turn ```"bash_arm_str" : 0.3,``` into e.g. ```[["bash", "arm", "str"], 0.3]```.  There might be a cleaner syntax you can use, but having it be slightly messier is way better than using a custom parser just for this field.